modulefiles/ufs_hera.intel

```diff
@@ -31,5 +31,5 @@ setenv CXX mpiicpc
 setenv FC mpiifort
 setenv CMAKE_Platform hera.intel
 
-# use shared memory and OpenFabrics Alliance (OFA) fabric with Intel MPI to circumvent RDMA-related bug in DAPL.
-setenv I_MPI_FABRICS shm:ofa
+# Intel MPI setting to circumvent RDMA-related bug in DAPL.
+setenv I_MPI_DAPL_UD 1
```

Comment:
Is this variable available on other platform too? Or is there change/updates required to use set this option on other platforms?

Reply:
Please remove this line too as it is deprecated as Intel 17. Hera admin suggested us not using it.

Reply:
@rmontuoro you okay with this?

Reply:
Setting `I_MPI_DAPL_UD` to 1 is required when running coupled GOCART with Intel MPI 2018.0.4. The model will crash otherwise.

Reply:
@rmontuoro @JessicaMeixner-NOAA  I saw 25 tracers are added in the cpld_bmark_aero test (compared to bmark test) and the number of nodes does not increase. and it takes 35 mins to finish. Since all the RT tests are required to finish within 30mins. I ran the test with 20 tasks/node (TPN_cpl_bmrk=20) and without setting I_MPI_DAPL_UD, the test finished in 24 mins. Since the ufs-weather-model supported platforms have different MPI versions, I think this might be helpful to port the aerosol code on other platforms without the MPI specific I_MPI setting.

Reply:
@junwang-noaa - Thank you for the suggestion. I've rerun the regression test using 20 tasks/node within 27 minutes of wall clock time on Hera, so I agree to setting `TPN_cpl_bmrk=20`.

My test also did not set `I_MPI_DAPL_UD`. Note that, as mentioned in previous discussions, setting `I_MPI_DAPL_UD` to 1 enables the connectionless DAPL UD transport, which is crucial when running the coupled model (w/ aerosols) on thousands of cores since the default transport would not scale sufficiently, causing timeout errors in the GOCART History component. This regression test uses only 560 MPI tasks and the default DAPL transport may still be adequate. Prototype runs, however, use well above 1k MPI tasks. In such cases, using the connectionless DAPL transport is highly recommended if not required to prevent communication failures.

Note also that the Intel MPI library _"switched from the Open Fabrics Alliance* (OFA) framework to the Open Fabrics Interfaces* (OFI) framework"_ with **release 2019**. Therefore, the DAPL fabric is deprecated **starting with the 2019 release** (see [Intel documentation](https://www.intel.com/content/www/us/en/developer/articles/technical/mpi-library-2019-over-libfabric.html)). The UFS weather model is using Intel MPI 2018 on Hera and Orion.

Reply:
I think it's OK to use DAPL, but the variable I_MPI_DAPL_UD is a depreciated option since Intel 17. Because of this, the impact on srun/slurm is unknown. Some of opn runs are using a couple of thousands tasks without using this setting. Do you have any test case with thousand tasks we can take look the performance of not using this setting? I am not sure what is the problem with those runs with thousand tasks

Reply:
@junwang-noaa - DAPL and `I_MPI_DAPL_UD` are deprecated since _**release 2019**_ of Intel MPI. We are still using Intel MPI 2018 on some platforms.

Setting `I_MPI_DAPL_UD` to 1 (as recommended by NASA) is necessary when running GOCART on a large number of MPI tasks, since the MAPL I/O implementation is based on MPI one-sided communication using Remote Memory Access (RMA) for higher performance. This requires a RDMA-capable communication fabric such as DAPL. Enabling DAPL connectionless transport (`I_MPI_DAPL_UD=1`) addresses scalability issues/failures at higher core counts and reduces the overall memory footprint. This setting is only needed when running the UFS weather model with GOCART.